ViAn/Filehandler/filehandler.cpp

```diff
@@ -115,6 +115,20 @@ void FileHandler::load_proj_files(std::string str){
 }
 
 /**
+ * @brief FileHandler::load_project
+ * @param dirpath
+ * @return
+ */
+Project* FileHandler::load_project(std::string fullProjectPath){
+    std::string reverse (fullProjectPath.rbegin(), fullProjectPath.rend());
+    std::string emaNjorp = reverse.substr(0, reverse.find("/"));
+    std::string projName (emaNjorp.rbegin(), emaNjorp.rend());
+    std::string dirpath = fullProjectPath.substr(0, fullProjectPath.find("/" + projName));
+    projName = projName.substr(0, projName.find(".txt"));
+    return load_project(projName, dirpath);
```

Comment:
If I understand this code correctly, what you want to do is separate the full path into a dirpath and projname, by splitting the string on the last /. There must be a simpler way to do this. Can't you use string::find_last_of or something similar. See [cplusplus.com](http://www.cplusplus.com/reference/string/string/find_last_of/) (they even have an example where they do exactly this).